src/Pandell.PackageRunner/Program.cs

```diff
@@ -215,17 +252,29 @@ private int Run(Assembly packageRunnerAssembly, string packageRunnerExeFileName,
             using (var catalog = new AggregateCatalog(assemblies))
             using (var container = new CompositionContainer(catalog))
             {
+                Program.WriteLine(verbose, "Resolving execution delegate.");
                 container.SatisfyImportsOnce(this);
                 if (this.RunAssembly == null)
                 {
                     Console.Error.WriteLine("Method not found!");
                     return 5;
                 }
+                Program.WriteLine(verbose, "Executing delegate.");
                 this.RunAssembly(configuration.token);
+                Program.WriteLine(verbose, "Delegate execution done.");
                 return 0;
             }
         }
 
+        /// <summary>
+        /// </summary>
+        private static void WriteLine(bool outputActive, string message, object[] args = null)
+        {
+            if (outputActive)
+            {
+                Console.WriteLine(message, args);
```

Comment:
Progress messages are typically written to `STDERR` (just like actual error messages), to separate this "program operation" stream from "program output" stream.

How about you create a delegate, e.g.

```c#
public delegate void WriteDiagnosticsMessage(TraceLevel messageLevel, string message, params object[] messageFormatParams);
```

- Implementation can format different levels differently (red for errors, yellow for warnings, etc).
- Implementation can include timestamp in standard format
- Implementation can be passed to execution package entry point so it reports progress in the same (consistent) way